aineko/core/dataset.py

```diff
@@ -56,33 +65,43 @@ def __init__(
         node_name: str,
         pipeline_name: str,
         dataset_config: Dict[str, Any],
-        broker: Optional[str] = None,
+        bootstrap_servers: Optional[str] = None,
+        prefix: Optional[str] = None,
+        has_pipeline_prefix: bool = False,
```

Comment:
Or to follow the Google Python Style Guide, we can do 

`if prefix is not None:`

Reply:
And then maybe raise a `ValueError` for empty strings?

Reply:
I'm intentionally creating 2 sets of prefixes: `prefix` and `pipeline_prefix`. 

`pipeline_prefix` is used for all user-defined datasets but not the `logging` topic. 

`prefix` is not currently used, but will be useful for adding a universal prefix to ALL datasets associated to this aineko project, so that one can namespace the kafka topics associated with this project. This will be required in a multi-tenant kafka setup, but not for a single-tenant setup.

Reply:
There's 2 levels of namespacing going on.
 
- Project-level namespace: defined by the `prefix`.
- Pipeline-level namespace: on by default, defined by `has_pipeline_prefix`